code/particle/util/ParticleProperties.cpp

```diff
@@ -37,21 +37,11 @@ void ParticleProperties::parse(bool nocreate) {
 int ParticleProperties::chooseBitmap()
 {
 	// failsafe check, though we should not have been able to get to this point
-	if (m_bitmap_list.empty())
-		return -1;
-
-	int bitmap_index;
-	int num_bitmaps = (int)m_bitmap_list.size();
-
-	if (num_bitmaps == 1) {
-		bitmap_index = m_bitmap_list[0];
-	} else if (num_bitmaps > 1) {
-		bitmap_index = m_bitmap_list[rand() % num_bitmaps];		
-	} else {
-		bitmap_index = -1;
-	}
+	Assertion(!m_bitmap_list.empty(), "No bitmaps found!");
+
+	size_t index = ::util::UniformRange<size_t>(0, m_bitmap_list.size() - 1).next();
 
-	return bitmap_index;
+	return m_bitmap_list[(int)index];
```

Comment:
Index operations can use `size_t` as their argument (it is actually the more correct type for that operation) so there is no need to have the int cast here.